app/models/role.rb

```diff
@@ -0,0 +1,39 @@
+# Model representing a user's role in a given course.
+class Role < ApplicationRecord
+  belongs_to :user
+  belongs_to :course
+  accepts_nested_attributes_for :user
+
+  # Group relationships
+  has_many :memberships, dependent: :delete_all
+  has_many :grade_entry_students
+  has_many :groupings, through: :memberships
+  has_many :notes, as: :noteable, dependent: :destroy
+  has_many :accepted_memberships,
```

Comment:
While we're at it, let's move this to the Student model. I really doesn't belong here